### DISCUSSION TITLE
Filtre géographique

### DISCUSSION THREAD

**Discussion by juliepierson** [geo2france/idg-qgis3-plugin#27] [General]
J'ouvre une discussion mais ça n'est pas pour qu'on garde absolument cette fonctionnalité, juste pour avoir un endroit où en discuter et garder trace de ce qui aura été décidé. Si vous pensez que ça a plus sa place dans les issues on peut le déplacer.

Donc cette fonctionnalité a été ajoutée [ici](https://gitlab.in2p3.fr/letg/indigeo-for-qgis/). Si on passe par le panneau explorateur plutôt que par un panneau dédié, on perd cette possibilité. J'ai posé hier la question de savoir si on pourrait intégrer un filtre géographique dans le panneau explorateur sur la liste qgis-developer, pas encore de réponse mais je pense que ce serait compliqué. Pour récupérer les emprises des couches WMS par exemple il faudrait faire des requêtes, ce qui reste d'être très long non ? Il peut y en avoir beaucoup ! Et idem pour les autres couches distantes.

Dans le panneau dédié ça reste possible puisque les emprises ont au préalable été récupérées dans le fichier de conf JSON (xmin, xmax, ymin et ymax).

Donc je pense qu'on perd cette fonctionnalité si on passe par le panneau explorateur. J'aurai tendance à dire que ça n'est pas dramatique et que ça fait partie du jeu !

Qu'en pensez-vous ?

**Comment by bchartier** [comment]
> [...]Pour récupérer les emprises des couches WMS par exemple il faudrait faire des requêtes, ce qui reste d'être très long non ? Il peut y en avoir beaucoup ! Et idem pour les autres couches distantes.

Pas sûr que ce soit très long. D'après ce que m'a dit @jbdesbas il compte utiliser une fonction de l'api QGIS pour charger un fichier projet. QGIS semble alors faire un certain nombre de vérification. Il est possible que l'emprise soit récupérée à ce moment-là soit parce que QGIS l'enregistre de toute manière dans le fichier projet soit parce qu'il récupèrerait des infos sur le service au moment du chargement (à voir/tester). D'ailleurs, si QGIS ne le fait pas de lui-même on pourrait peut-être imaginer stocker cette information dans la partie "métadonnées / emprise" de la fenêtre de propriété de chaque couche. Si cette info est renseignée on pourrait imaginer l'exploiter. Mais c'est pas non plus super simple.

> Dans le panneau dédié ça reste possible puisque les emprises ont au préalable été récupérées dans le fichier de conf JSON (xmin, xmax, ymin et ymax).

Oui ça reste possible dans le cas des JSON. À voir pour les projet QGIS.

**Reply by juliepierson** [reply]
Merci Benjamin pour ta réponse. Je ne sais pas si je loupe un truc, mais s'il y avait un filtre géographique dans l'explorateur il devrait fonctionner sur toutes les couches pour lesquelles l'utilisateur a une connexion, et pas seulement celles du plugin ?

**Reply by bchartier** [reply]
Je ne pense pas que tu loupes un truc. Cela dépend ce que l'on décide de faire je pense :
* Si on ajoute un filtre géographique dans le cadre du plugin il n'y a aucune raison qu'il fonctionne pour les connexions non gérées par le plugin.
* Si on se débrouille pour ajouter un filtre géographique au cœur de QGIS effectivement il pourrait s'appliquer à toutes les connexions.

Maintenant, quand on voit comment fonctionne le filtre textuel de l'explorateur QGIS, il n'a pas l'air de s'appliquer à toutes les connexions. J'ai l'impression qu'il ne s'applique qu'aux branches qui ont été ouvertes par l'utilisateur. Enfin, c'est ce que j'ai retenu du test qu'on a fait rapidement avec @jbdesbas hier.
